diff --git a/src/libflake/flake.cc b/src/libflake/flake.cc
index 8e7e2be26..5d8834668 100644
--- a/src/libflake/flake.cc
+++ b/src/libflake/flake.cc
@@ -420,7 +420,7 @@ lockFlake(const Settings & settings, EvalState & state, const FlakeRef & topRef,
 
     auto useRegistries = lockFlags.useRegistries.value_or(settings.useRegistries);
     auto useRegistriesTop = useRegistries ? fetchers::UseRegistries::All : fetchers::UseRegistries::No;
-    auto useRegistriesInputs = useRegistries ? fetchers::UseRegistries::Limited : fetchers::UseRegistries::No;
+    auto useRegistriesInputs = useRegistries ? fetchers::UseRegistries::All : fetchers::UseRegistries::No;
 
     auto flake = getFlake(state, topRef, useRegistriesTop, {});
 
diff --git a/tests/functional/flakes/flakes.sh b/tests/functional/flakes/flakes.sh
index 5b1da0f02..7ac968e24 100755
--- a/tests/functional/flakes/flakes.sh
+++ b/tests/functional/flakes/flakes.sh
@@ -226,7 +226,8 @@ mv "$registry.tmp" "$registry"
 mkdir -p "$TEST_HOME/.config/nix"
 ln -sfn "$registry" "$TEST_HOME/.config/nix/registry.json"
 nix flake metadata --flake-registry '' flake1
-expectStderr 1 nix flake update --flake-registry '' --flake "$flake3Dir" | grepQuiet "cannot find flake 'flake:flake1' in the flake registries"
+nix flake update --flake-registry '' --flake "$flake3Dir"
+[[ $(jq -c .nodes.flake1.original.type "$flake3Dir/flake.lock") =~ 'indirect' ]]
 rm "$TEST_HOME/.config/nix/registry.json"
 
 # Test whether flakes are registered as GC roots for offline use.
